test/spark.js

```diff
@@ -0,0 +1,70 @@
+import Spark from '../lib/spark.js'
+import { test } from 'zinnia:test'
+import { assertInstanceOf, assertEquals } from 'zinnia:assert'
+
+test('getRetrieval', async () => {
+  const retrieval = { retrieval: 'retrieval' }
+  const fetch = async (url, opts) => {
+    assertEquals(url, 'https://spark.fly.dev/retrievals')
+    assertEquals(opts.method, 'POST')
+    return {
+      async json () {
+        return retrieval
+      }
+    }
+  }
+  const spark = new Spark({ fetch })
+  assertEquals(await spark.getRetrieval(), retrieval)
+})
+
+// TODO: test more cases
+test('fetchCAR', async () => {
+  const URL = 'url'
+  const fetch = async url => {
+    assertEquals(url, URL)
+    return {
+      status: 200,
+      ok: true,
+      body: {
+        [Symbol.asyncIterator]: () => {
+          let i = 0
+          return {
+            async next () {
+              if (i === 0) {
+                i++
+                return {
+                  value: new Uint8Array([1, 2, 3]),
+                  done: false
+                }
+              } else {
+                return {
+                  value: undefined,
+                  done: true
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
```

Comment:
Have you considered using async generator function instead of implementing the AsyncIterator protocol manually?

I am not sure what type is returned by async generator functions, I am assuming something like this may work:

```js
body: (async function* () {
  yield new Uint8Array([1, 2, 3]);
  yield new Uint8Array([4, 5, 6 ]);
})()
```

See also:
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/async_function*
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AsyncGeneratorFunction

Reply:
Good idea, refactored like this 👍